src/common/libkvs/treeobj.c

```diff
@@ -162,20 +162,20 @@ int treeobj_decode_val (json_t *obj, void **dp, int *lp)
     }
     xdatastr = json_string_value (xdata);
     xlen = strlen (xdatastr);
-    len = base64_decode_length (xlen);
-    if (len > 1) {
-        if (!(data = malloc (len))) {
-            errno = ENOMEM;
-            return -1;
-        }
-        if (base64_decode_block (data, &len, xdatastr, xlen) < 0) {
-            free (data);
-            errno = EINVAL;
-            return -1;
-        }
+    buflen = base64_decode_length (xlen);
+    assert (buflen > 0);
+    if (!(data = malloc (buflen)))
+        return -1;
+    len = buflen;
+    if (base64_decode_block (data, &len, xdatastr, xlen) < 0) {
+        free (data);
+        errno = EINVAL;
+        return -1;
     }
-    else {
-        len = 0;
+    assert (len < buflen);
```

Comment:
These asserts confused me a bit, so perhaps a comment would be useful.  It seems that ```base64_decode_length()``` always includes extra buffer space?  If the asserts were:

```
assert (len < buflen);
assert (data[len + 1] == '\0');
```
or

```
assert (len <= buflen);
assert (data[len] == '\0');
```

I think it would make more sense (the former would make more sense given the ```len == 0``` check below.

Reply:
oops, the ***latter*** would make more sense given the ```len == 0``` check below.

Reply:
Re-reading this, what I said may have been confusing b/c I was dumping thoughts.  Let me re-write.

These asserts confused me a bit, so perhaps a comment would be useful.  If the assert was

```
assert (len <= buflen);
assert (data[len] == '\0');
```

it would make sense.  Perhaps ```base64_decode_length()``` always includes extra buffer space?  That would make the present asserts make sense.

Reply:
I'll add some comments - I think the assertions themselves are probably right.  Have a look in a sec when I push an update.

Reply:
ahhh, ```len``` doesn't include the '\0' term.  Now it makes more sense.